src/core/ext/filters/client_channel/retry_throttle.cc

```diff
@@ -119,42 +120,41 @@ void ServerRetryThrottleData::RecordSuccess() {
 // ServerRetryThrottleMap
 //
 
-using StringToDataMap =
-    std::map<std::string, RefCountedPtr<ServerRetryThrottleData>>;
-static gpr_mu g_mu;
-static StringToDataMap* g_map;
-
-void ServerRetryThrottleMap::Init() {
-  gpr_mu_init(&g_mu);
-  g_map = new StringToDataMap();
-}
-
-void ServerRetryThrottleMap::Shutdown() {
-  gpr_mu_destroy(&g_mu);
-  delete g_map;
-  g_map = nullptr;
+std::atomic<ServerRetryThrottleMap*> ServerRetryThrottleMap::instance_{nullptr};
+
+ServerRetryThrottleMap* ServerRetryThrottleMap::Get() {
+  ServerRetryThrottleMap* p = instance_.load(std::memory_order_acquire);
+  if (p == nullptr) {
+    p = new ServerRetryThrottleMap();
+    ServerRetryThrottleMap* expected = nullptr;
+    if (!instance_.compare_exchange_strong(expected, p,
+                                           std::memory_order_acq_rel,
+                                           std::memory_order_acquire)) {
+      delete p;
+      p = expected;
+    }
+  }
+  return p;
 }
 
 RefCountedPtr<ServerRetryThrottleData> ServerRetryThrottleMap::GetDataForServer(
     const std::string& server_name, intptr_t max_milli_tokens,
     intptr_t milli_token_ratio) {
   RefCountedPtr<ServerRetryThrottleData> result;
```

Comment:
Not related to this PR, but doesn't look like this variable is actually used, so I think it can be removed.

Reply:
Done